### DISCUSSION TITLE
What version of Unity should we upgrade to?

### DISCUSSION THREAD

**Discussion by cosmiccoincidence** [RE-SS3D/SS3D#720] [Q&A] [answered]
This discussion is linked to Issue #655 
It's highly recommended that we upgrade to an LTS version for various reasons, the question is which.

Below are our options in the form of a comment chain for each option. I'd like for you to comment on them with the good and bad things regarding using that option. Also, vote on your preferred option (with the little up arrows to the left of the comments) and maybe add a comment with your specific reasoning.

**Comment by cosmiccoincidence** [comment, marked as answer]
### **2019 LTS**
(with the same features as current version, but more stable)

**Reply by Ryan089** [reply]
This version will require the least changes to our existing code. It is also the most supported by Mirror, as per the Recommended Unity Version on Mirror documentation: "In fact if you care about stability, sticking with the previous LTS version (2019 LTS) is a good idea."

**Reply by stilnat** [reply]
I'll stick with Mirror's recommendation for now, even though the new profiling tool on 2020 looks sweet.

**Reply by cosmiccoincidence** [reply]
Decided on 2019 LTS for now.

**Comment by cosmiccoincidence** [comment]
### **2020 LTS**
(more features than current version, but less stable)
https://unity3d.com/unity/whats-new/2020.3.12
https://unity.com/releases/2020-1

**Reply by Ryan089** [reply]
This option is alright, but will be more work than upgrading to 2019 LTS.

**Reply by TaylorNAlbarnaz** [reply]
This version also will give us access to spatial hashing which will be really useful for the Tilemap and possibly Atmos, so it has both pros and cons

**Reply by stilnat** [reply]
Spatial Hashing is also supported for Unity 2019 LTS

**Comment by cosmiccoincidence** [comment]
### **2021 LTS**
(even features, but likely even less stable)
https://unity3d.com/unity/whats-new/2021.1.0
https://blog.unity.com/technology/the-road-to-2021

**Reply by cosmiccoincidence** [reply]
A negative with this is that 2021 LTS is not released yet and going with this option means we will have to wait 4-5 months until it is released before we update.

**Reply by Ryan089** [reply]
This option is terrible. Upgrading Unity is on a critical path for future RE:SS3D development, and delaying the Unity upgrade will prevent most further meaningful work from being done for half a year. Critical path:
- Upgrading Unity is a prerequisite for upgrading Mirror version.
- Upgrading Mirror version is a prerequisite for using SpatialHashingInterestManagement, which I consider as an essential part of the solution for tilemap networking (and in fact, networking in general).
- Correct Tilemap networking is essential for pretty much every other significant system in the game.

However, I don't have an objection to upgrading to 2019 LTS as an interim measure so that we can progress the game code, then upgrading to 2021 LTS when it comes out.

**Comment by Birlinha** [comment]
I wanted to add to the (now closed) discussion with 

- Why Unity, at all?
Unity is a very popular open source game development engine, with years of existance in the market
Because of it's popularity, a lot of people will instantly recognize their engine, even if the version is a little bit different

This question came into my mind because I was developing a ~useless~ instance of SS13 3D, which you can check here: https://playcanvas.com/project/825830/overview/move
This website is very similar to Unity, it is a game development engine
But it is different in 2 aspects:
- Cloud Engine (you can program even if you don't have a Personal Computer - it is very easy to open a JS file and tweak it)
- Cross-Platform reach (playable across platforms, through browser - then, you just have to adapt each control the players might have)

Are there other engines I'm forgetting?
